integrations/gitlab/gitlab_integration/ocean.py

```diff
@@ -207,3 +212,21 @@ async def resync_pipelines(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
                         {**pipeline.asdict(), "__project": project.asdict()}
                         for pipeline in pipelines_batch
                     ]
+
+
+@ocean.on_resync(ObjectKind.MEMBER)
+async def resync_members(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    gitlab_resource_config: GitlabMembersResourceConfig = typing.cast(
+        GitlabMembersResourceConfig, event.resource_config
+    )
+    selector = gitlab_resource_config.selector
+    for service in get_cached_all_services():
+        for group in service.get_root_groups():
+            async for members in service.get_all_group_members(group):
+                if selector.public_email_visibility:
+                    yield [
+                        await service.enrich_member_with_public_email(member)
+                        for member in members
+                    ]
+                else:
+                    yield [member.asdict() for member in members]
```

Comment:
what happens when I have the same user in multiple groups? how would that behave? will I have to perform repeated upserts?

maybe in this method ^ we should use `members = group.members_all.list(get_all=True)` which will return all and reduce the amount of extra requests that we will have to perform?

Reply:
my reason for using members as opposed to member_all was because the members_all request returns not the the user in that group but also all inherited and invited members.

Thereby resulting in all groups the same members since the members most commonly belong to the parent group - [details](https://docs.gitlab.com/ee/user/group/subgroups/#:~:text=Inherited%20members%20of%20the%20subgroup,the%20subgroup%20or%20its%20ancestors.)

aside this, the behavior and how we will retrieve members does not differ from member and members_all

Reply:
ok i understand what you are saying, so if we use the `members_all` only for the members kind wouldn't it reduce the amount of requests by a lot? as we will only have to bring the members for the root groups rather than the `subgroups` as well. 

also just making sure that you have tested subgroups as well. please confirm

Reply:
calling `/members` on root groups returns the same results as `/members/all`, calling `/members` on subgroups comes with less data than `members/all`, due to the inclusion of invited and inherited members, for root groups, concept of inherited members does not apply, all members of the root groups are returned regardless how we choose to call them.

I believe the optimization here was getting members from root groups instead of all groups (including subgroups), which would have taken more time.